ev3dev2/sensor/lego.py

```diff
@@ -440,12 +440,12 @@ def __init__(self, address=None, name_pattern=SYSTEM_DEVICE_NAME_CONVENTION, nam
 
     @property
     def distance_centimeters_continuous(self):
-        self.mode = self.MODE_US_DIST_CM
+        self._ensure_mode(self.MODE_US_DIST_CM)
         return self.value(0) * self._scale('US_DIST_CM')
 
     @property
     def distance_centimeters_ping(self):
-        self.mode = self.MODE_US_SI_CM
+        self._ensure_mode(self.MODE_US_SI_CM)
```

Comment:
This mode is special. The value does not update unless you set the mode. So this change will cause the sensor the become unresponsive after the first call.

Reply:
interesting, it also appears that this mode has a problem with locking up the sensor if you poll it too often. http://docs.ev3dev.org/projects/lego-linux-drivers/en/ev3dev-stretch/sensor_data.html#lego-ev3-us-mode3-value0

It would be worth copying that note to the python docs if it isn't there already

Reply:
Thanks; that's a very good point. I'll revert these two back to just setting the mode directly and copy over the note.